src/ts/waterfall/row/svg-indicators.ts

```diff
@@ -30,47 +30,48 @@ export function getMimeTypeIcon(entry: WaterfallEntry): Icon {
   if (!!harEntry.response.redirectURL) {
     const url = encodeURI(harEntry.response.redirectURL.split("?")[0] || "");
     return makeIcon("err3xx", `${harEntry.response.status} response status: Redirect to ${url}...`);
-  } else if (heuristics.isInStatusCodeRange(harEntry, 400, 499)) {
+  } else if (isInStatusCodeRange(harEntry, 400, 499)) {
     return makeIcon("err4xx", `${harEntry.response.status} response status: ${harEntry.response.statusText}`);
-  } else if (heuristics.isInStatusCodeRange(harEntry, 500, 599)) {
+  } else if (isInStatusCodeRange(harEntry, 500, 599)) {
     return makeIcon("err5xx", `${harEntry.response.status} response status: ${harEntry.response.statusText}`);
   } else if (harEntry.response.status === 204) {
     return makeIcon("plain", "No content");
   } else {
     return makeIcon(entry.requestType, entry.requestType);
   }
 }
-  /**
-   * Scan the request for errors or portential issues and highlight them
-   * @param  {WaterfallEntry} entry
-   * @param  {boolean} docIsSsl
-   * @returns {Icon[]}
-   */
-  export function getIndicatorIcons(entry: WaterfallEntry, docIsSsl: boolean): Icon[] {
-  const harEntry = entry.rawResource;
-  let output = [];
-
-  if (heuristics.isPush(entry)) {
-    output.push(makeIcon("push", "Response was pushed by the server"));
+/**
+ * Gets the Indicators in Icon format
+ * @param  {WaterfallEntry} entry
+ * @returns {Icon[]}
+ */
+export function getIndicatorIcons(entry: WaterfallEntry): Icon[] {
+  if (entry.indicators.length === 0) {
+    return [];
   }
 
-  if (docIsSsl && !heuristics.isSecure(entry)) {
-    output.push(makeIcon("noTls", "Insecure Connection"));
-  }
+  let combinedTitle = [];
+  let icon = "";
+  const errors = entry.indicators.filter((i) => i.type === "error");
+  const warnings = entry.indicators.filter((i) => i.type === "warning");
+  const info = entry.indicators.filter((i) => i.type !== "error" && i.type !== "warning");
 
```

Comment:
Show the icon with the highest priority `error` > `warning` > `info`||customIcon.

The icons title shows all `icon.titles` combined and grouped by type.